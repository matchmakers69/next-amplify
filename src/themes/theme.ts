import baseStyled, { ThemedStyledInterface, DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  title: "theme",
  colors: {
    bodyColor: "#f9f1e4",
    footerColor: "#ebc68f",
    shapeColor: "#f6e1c0",
    white: "#ffffff",
    persianGreen: "#daee73",
    text: "#010000",
    yellow: "#ead96d",
    grey100: "#e8e8e0",
    lightBrown: "#ecdcc2",
    red: "#c0362b",
    logoLightGreen: "#739c3e",
    logoDarkGreen: "#1c2e0d",
    mustard: "#bb8918",
    darkMustard: "#ba8d0f",
    titleGreen: "#2b331f",
    fadeGreen: "#afb3aa",
    mustardLabel: "#af7f0d",
    lightBlue: "#2193a2",
    transparent: "transparent"
  },

  fontSizes: {
    xs: "1.4rem",
    sm: "1.6rem",
    md: "2rem",
    large: "3rem"
  },
  fontFamily: {
    primary: "Poppins",
    secondary: "Roboto Slab"
  },
  weight: {
    thin: 300,
    regular: 400,
    medium: 500,
    semiBold: 600,
    bold: 700
  }
};
export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
