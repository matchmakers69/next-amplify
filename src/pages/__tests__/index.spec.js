import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { theme } from 'src/themes/theme';
import Home from '..';

describe('<Home />', () => {
  it('should create a snapshot for Home Page', () => {
    expect(
      renderer
        .create(
          <StyledThemeProvider theme={theme}>
            <Home />
          </StyledThemeProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
  it('renders Home page without crashing', () => {
    const { getByTestId } = render(
      <StyledThemeProvider theme={theme}>
        <Home />
      </StyledThemeProvider>,
    );

    const pageWrapper = getByTestId('page-wrapper');
    expect(pageWrapper).toBeInTheDocument();
  });
});
