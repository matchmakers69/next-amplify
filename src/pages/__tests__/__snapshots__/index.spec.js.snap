// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should create a snapshot for Home Page 1`] = `
<div
  data-testid="page-wrapper"
>
  Home Page
</div>
`;
